src/Waives.Pipelines/WaivesDocument.cs

```diff
@@ -46,6 +47,14 @@ public async Task<WaivesDocument> Extract(string extractorName)
             };
         }
 
+        public async Task<WaivesDocument> Redact(string extractorName, Func<WaivesDocument, Stream, Task> resultFunc)
+        {
+            var resultStream = await _waivesDocument.Redact(extractorName);
+            await resultFunc(this, resultStream);
```

Comment:
:hand: I think we need to guard against `resultFunc` being `null` here

Reply:
Resolved in 38c4218.